lib/src/widgets/scroll_to_bottom_button.dart

```diff
@@ -0,0 +1,95 @@
+import 'package:chatview/src/extensions/extensions.dart';
+import 'package:flutter/material.dart';
+
+class ScrollToBottomButton extends StatefulWidget {
+  const ScrollToBottomButton({super.key});
+
+  @override
+  ScrollToBottomButtonState createState() => ScrollToBottomButtonState();
+}
+
+class ScrollToBottomButtonState extends State<ScrollToBottomButton> {
+  bool _isZoomedIn = false;
+  late ScrollController? scrollController;
+
+  @override
+  void initState() {
+    super.initState();
+
+    WidgetsBinding.instance.addPostFrameCallback((_) {
+      scrollController = chatViewIW?.chatController.scrollController;
+      scrollController?.addListener(_updateScrollButtonVisibility);
+    });
+  }
+
+  void _updateScrollButtonVisibility() {
+    final scrollToBottomButtonConfig =
+        chatListConfig.scrollToBottomButtonConfig;
+    if ((scrollController?.offset ?? 0) >
+        (scrollToBottomButtonConfig?.buttonDisplayOffset ?? 300)) {
+      if ((!_isZoomedIn) && mounted) {
+        setState(() {
+          _isZoomedIn = true;
+        });
+      }
+    } else {
+      if (_isZoomedIn && mounted) {
+        setState(() {
+          _isZoomedIn = false;
+        });
+      }
+    }
+  }
```

Comment:
I guess this can be refactored with the below code. What's your thought?

```
  void _updateScrollButtonVisibility() {
    if (!mounted) return;

    final double currentOffset = scrollController?.offset ?? 0;
    final double buttonDisplayOffset =
        chatListConfig.scrollToBottomButtonConfig?.buttonDisplayOffset ?? 300;
    final bool shouldZoomIn = currentOffset > buttonDisplayOffset;

    setState(() {
      _isZoomedIn = shouldZoomIn;
    });
  }
```

Reply:
we can't use this as setState will be triggered for each listener call which is not required we only want it to be triggered only when it passes the offset and bool is different then it should be.